node_modules/col-activities/lib/notifications/weekly.js

```diff
@@ -433,32 +447,59 @@ var getUserTotals = function(userId, summary) {
  * Running totals for activites and points per asset.
  */
 var AssetTotals = function() {
-  _.extend(this, {
+  var that = {
     'comments': 0,
     'likes': 0,
     'views': 0
-  });
-}
+  };
+
+  /**
+   * Given an activity type, find the appropriate running-total property and increment by one.
+   *
+   * @param  {String}    activityType     The activity type to be translated
+   */
+  that.incrementActivities = function(activityType) {
```

Comment:
We can make `incrementActivities` a reusable function, as it is the same for all classes:

```
var AssetTotals = function() {
  var that = {
    'comments': 0,
    'likes': 0,
    'views': 0
  };

  that.dictionary = {
    'asset_comment': 'comments',
    'like': 'likes',
    'view_asset': 'views'
  };

  that.incrementActivities = getActivitiesIncrementer(that);
```

or 

```
var AssetTotals = function() {
  var that = {
    'comments': 0,
    'likes': 0,
    'views': 0
  };

  var dictionary = {
    'asset_comment': 'comments',
    'like': 'likes',
    'view_asset': 'views'
  };

  that.incrementActivities = getActivitiesIncrementer(that, dictionary);
```